elitzur-core/src/main/scala/com/spotify/elitzur/validators/Validator.scala

```diff
@@ -271,47 +273,52 @@ object Validator extends Serializable {
         if (!accessor.validator.isInstanceOf[IgnoreValidator[_]]) {
           val name = new JStringBuilder(path.length + accessor.label.length)
             .append(path).append(accessor.label).toString
-          val o = if (accessor.validator.isInstanceOf[FieldValidator[_]]) {
+          val c = config.fieldConfig(name)
+          val (o, validationType) = if (accessor.validator.isInstanceOf[FieldValidator[_]]) {
             val v = accessor.validator.validateRecord(Unvalidated(accessor.value))
             val validationType = accessor.validator.asInstanceOf[FieldValidator[_]].validationType
+            (v, Some(validationType))
+          } else {
 
-            val c = config.fieldConfig(name)
-            if (v.isValid) {
-              if (c != NoCounter) {
-                reporter.reportValid(
-                  outermostClassName,
-                  name,
-                  validationType
-                )
-              }
-            } else if (v.isInvalid) {
-              if (c == ThrowException) {
-                throw new DataInvalidException(
-                  s"Invalid value ${v.forceGet.toString} found for field $path${accessor.label}")
+            val validationType =
```

Comment:
Similarly we can change this to just mutating a var instead of having the None clause